File: ProjOOPJava/less6/controller/DeliveryController.java
package ProjOOPJava.less6.controller;
import ProjOOPJava.less6.model.*;
import ProjOOPJava.less6.service.DelivetyEnum;

public class DeliveryController {
    private DeliveryAction action;
    public void delivery(DelivetyEnum type){
        action = DeliveryBuild.getTransport(type);
    }
    public String getTransport(){
        return action.getTransport();
    }
    public String receivingAnOrder(){
        return action.receivingAnOrder();
    }
    public String deliveryOrder(){
        return action.deliveryOrder();
    }
    public String deliveryComplite(){
        return "Доставка выполнена";
    }
}

File: ProjOOPJava/less7/controller/DeliveryController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WinFormsApp1.model;
using WinFormsApp1.service;

namespace WinFormsApp1.controller
{
    internal class DeliveryController
    {
        private DeliveryAction action;
        public void delivery(DeliveryEnum type)
        {
            action = DeliveryBuild.getTransport(type);
        }
        public String getTransport()
        {
            return ((DeliveryGet)action).getTransport();
        }
        public String receivingAnOrder()
        {
            return action.receivingAnOrder();
        }
        public String deliveryOrder()
        {
            return action.deliveryOrder();
        }
        public String deliveryComplite()
        {
            return "Доставка выполнена";
        }
    }
}
